From 5d2e8ad3812642bfbe7eb3df7117590b9b20450a Mon Sep 17 00:00:00 2001
From: Oliver Neukum <oneukum@suse.com>
Date: Wed, 18 Mar 2020 14:34:56 +0100
Subject: [PATCH] kABI: more hooks for PCI changes

Patch-mainline: never, kABI guards
References: bsn#129 bsc#1141558
Modified-by-SEL: Yes, modified due to different context

This bug may do extensive surgery in the future, adding maximum hooks

Signed-off-by: Oliver Neukum <oneukum@suse.de>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/pci/hotplug/pciehp.h |    2 ++
 include/linux/pci.h          |    6 ++++++
 2 files changed, 8 insertions(+)

--- a/drivers/pci/hotplug/pciehp.h
+++ b/drivers/pci/hotplug/pciehp.h
@@ -109,6 +109,8 @@ struct controller {
 	unsigned int ist_running;
 	int request_result;
 	wait_queue_head_t requester;
+
+	void* suse_kabi_padding;
 };
 
 /**
--- a/include/linux/pci.h
+++ b/include/linux/pci.h
@@ -73,6 +73,8 @@ struct pci_slot {
 	struct hotplug_slot	*hotplug;	/* Hotplug info (move here) */
 	unsigned char		number;		/* PCI_SLOT(pci_dev->devfn) */
 	struct kobject		kobj;
+
+	void* suse_kabi_padding;
 };
 
 static inline const char *pci_slot_name(const struct pci_slot *slot)
@@ -511,6 +513,8 @@ struct pci_dev {
 	size_t		romlen;		/* Length if not from BAR */
 	char		*driver_override; /* Driver name to force a match */
 
+	void* suse_kabi_padding;
+
 	unsigned long	priv_flags;	/* Private flags for the PCI driver */
 	/*
 	 * This flag is only set on root ports. When a slot below a root port
@@ -580,6 +584,8 @@ struct pci_host_bridge {
 	unsigned int	preserve_config:1;	/* Preserve FW resource setup */
 	unsigned int	size_windows:1;		/* Enable root bus sizing */
 
+	void* suse_kabi_padding;
+
 	/* Resource alignment requirements */
 	resource_size_t (*align_resource)(struct pci_dev *dev,
 			const struct resource *res,
